lib/spack/spack/spec.py

```diff
@@ -572,35 +581,26 @@ def __init__(self, *args):
             self.versions.add(versions)
 
         else:
-            raise TypeError("__init__ takes 1 or 2 arguments. (%d given)" % nargs)
-
-    def _add_versions(self, version_list):
-        # If it already has a non-trivial version list, this is an error
-        if self.versions and self.versions != vn.VersionList(":"):
-            # Note: This may be impossible to reach by the current parser
-            # Keeping it in case the implementation changes.
-            raise MultipleVersionError(
-                "A spec cannot contain multiple version signifiers." " Use a version list instead."
-            )
-        self.versions = vn.VersionList()
-        for version in version_list:
-            self.versions.add(version)
+            raise TypeError(f"__init__ takes 1 or 2 arguments. ({nargs:d} given)")
 
-    def _autospec(self, compiler_spec_like):
+    @staticmethod
+    def _maybe_literal(compiler_spec_like: Union[str, "CompilerSpec"]) -> "CompilerSpec":
```

Comment:
Why do we need this if we can just add another overload to the constructor that admits a `CompilerSpec` arg?

Reply:
This is keeping the same behavior we had before. Using the constructor creates a copy, in this way we avoid it. Not sure it matters for performance, but I thought that was the reason we added `_autospec`